examples/do_MLICA.py

```diff
@@ -0,0 +1,134 @@
+
+ica = ICA(method='fastica', n_components=n_components, random_state=None,
+          max_pca_components=None, max_iter=1000, verbose=None)
+
+# do the ICA decomposition on downsampled raw
+ica.fit(raw_ds_chop, picks=picks, reject={'mag': 5e-12},
+        verbose=None)
+
+sources = ica.get_sources(raw_ds_chop)._data
+
+# extract temporal and spatial components
+mm = np.float32(np.dot(ica.mixing_matrix_[:, :].T,
+                       ica.pca_components_[:ica.n_components_]))
+
+# use [:, :15000] to make sure it's 15000 data points
+chop = sources[:, :15000]
+chop_reshaped = np.reshape(chop, (len(chop), len(chop[0]), 1))
+
+model_scores = model.predict([mm, chop_reshaped], verbose=1)
+
+bads_MLICA = []
+
+print model_scores
+
+for idx in range(0, len(model_scores)):
+    if model_scores[idx][0] > MLICA_threshold:
+        bads_MLICA.append(idx)
+
+ica.exclude = bads_MLICA
+
+# visualisation
+ica.plot_sources(raw_ds_chop, block=True)
+
+# compare MLICA to results from correlation and ctps analysis
+ica.exclude = []
+
+print 'Computing scores and identifying components..'
+ecg_scores = ica.score_sources(raw_chop, target=ecg_ch, score_func='pearsonr',
+                               l_freq=flow_ecg, h_freq=fhigh_ecg, verbose=False)
+# horizontal channel
+eog1_scores = ica.score_sources(raw_chop, target=eog1_ch, score_func='pearsonr',
+                                l_freq=flow_eog, h_freq=fhigh_eog, verbose=False)
+# vertical channel
+eog2_scores = ica.score_sources(raw_chop, target=eog2_ch, score_func='pearsonr',
+                                l_freq=flow_eog, h_freq=fhigh_eog, verbose=False)
+
+# print the top ecg, eog correlation scores
+ecg_inds = np.where(np.abs(ecg_scores) > ecg_thresh)[0]
+eog1_inds = np.where(np.abs(eog1_scores) > eog_thresh)[0]
+eog2_inds = np.where(np.abs(eog2_scores) > eog_thresh)[0]
+highly_corr = list(set(np.concatenate((ecg_inds, eog1_inds, eog2_inds))))
+print 'Highly correlated artifact components are', highly_corr
+
+
+bads_corr_ctps = []
+bads_corr_ctps += highly_corr
+
+# get ECG/EOG related components using JuMEG
+ic_ecg = get_ics_cardiac(raw_chop, ica,
+                         flow=flow_ecg, fhigh=fhigh_ecg,
+                         thresh=ecg_thresh,
+                         tmin=-0.5, tmax=0.5, name_ecg=ecg_ch,
+                         score_func='pearsonr',
+                         use_CTPS=True)
+ic_eog = get_ics_ocular(raw_chop, ica,
+                        flow=flow_eog, fhigh=fhigh_eog, thresh=eog_thresh,
+                        name_eog_hor=eog1_ch, name_eog_ver=eog2_ch,
+                        score_func='pearsonr')
+
+bads_corr_ctps += list(ic_ecg) + list(ic_eog)
+bads_corr_ctps = list(set(bads_corr_ctps))
+bads_corr_ctps.sort()
+ica.exclude = bads_corr_ctps  # to sort and remove repeats
+# visualisation
+ica.plot_sources(raw_chop, block=True)
+
+print 'Bad components from MLICA:', bads_MLICA
+print 'Bad components from correlation & ctps:', ica.exclude
```

Comment:
We could also extend the example to apply ica on unfiltered data using the latest jumeg function. Then once the ica is applied, we plot the performance plots of cleaned data for MLICA.